src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy

```diff
@@ -794,20 +801,20 @@ public enum LegendsAwakened implements LogicCardInfo {
           weakness P, '+20'
           pokePower "Time Walk", {
             text "Once during your turn, when you put Azelf from your hand onto your Bench, you may look at all of your face-down Prize cards. If you do, you may choose 1 Pokémon you find there, show it to your opponent, and put it into your hand. Then, choose 1 card in your hand and put it as a Prize card face down."
-            actionA {
-              assert my.hand : "No cards in hand"
-              checkLastTurn()
-              powerUsed()
-
-              def newPrize = my.hand.select(hidden: true, "Card to put into Prizes").first()
-
-              def tar = my.prizeCardSet.faceDownCards.select(hidden: false, "Choose a Prize card to replace with one in your hand.").first()
-              my.hand.add(tar)
-
-              def indexOfOldPrize = my.prizeCardSet.indexOf(tar)
-              my.prizeCardSet.set(indexOfOldPrize, newPrize)
-              my.prizeCardSet.setVisible(newPrize, true)
-              my.hand.remove(newPrize)
+            onActivate {r->
+              if (r==PLAY_FROM_HAND && confirm("Use Time Walk?")){
+                assert my.hand : "No cards in hand"
+                powerUsed()
+                def tar = my.prizeCardSet.faceDownCards.select(hidden: false, min: 0, "Choose a Pokemon in your prizes to replace with a card in your hand.").first()
+                assert tar : "No Pokemon selected."
+                assert tar.hasTypes(POKEMON) : "Only Pokemon can be grabbed by Time Walk."
```

Comment:
`assert`s are inappropriate after `powerUsed()` has been called. They are to prevent attacks and abilities from running, not for ending them prematurely.